armory/data/datasets.py

```diff
@@ -496,6 +498,19 @@ def librispeech_dev_clean(
             "Processing of variable length inputs not yet implemented."
         )
 
+    if not dataset_dir:
+        dataset_dir = paths.docker().dataset_dir
+
+    if cache_dataset and not os.path.isdir(
+        os.path.join(dataset_dir, "librispeech_dev_clean_split", "plain_text", "1.1.0")
+    ):
+        download_verify_dataset_cache(
+            dataset_dir=dataset_dir,
+            checksum_file=os.path.join(
+                CACHED_CHECKSUMS_DIR, "librispeech_dev_clean_split.txt"
+            ),
+        )
+
```

Comment:
Instead of duplicating this for every dataset of interest, can you create a single function to handle this? Something that you could call from here like:
```python
if cache_dataset:
    _cache_dataset_function(dataset_dir, name="librispeech_dev_clean_split", subpath=os.path.join("plain_text", "1.1.0"))
```